skrub/tests/test_deduplicate.py

```diff
@@ -95,3 +98,67 @@ def test__create_spelling_correction(seed: int = 123) -> None:
             spelling_correction.values[clusters == n].astype("int")
             == counts[clusters == n].max()
         ).all()
+
+
+@skip_if_no_parallel
+def test_parallelism():
+    # Test that parallelism works
+    X = make_deduplication_data(examples=['black', 'white', 'red'], entries_per_example=[500, 500, 500], prob_mistake_per_letter=0.3)
+    y = deduplicate(X, n_jobs=None)
+    times = {}
+    jobs = [1, 4]
+
+    for n_jobs in jobs:
+        start = time.time()
+        y_parallel = deduplicate(X, n_jobs=n_jobs)
+        assert_array_equal(y, y_parallel)
+        end = time.time()
+        timer = end - start
+        times[n_jobs] = timer
```

Comment:
When measuring time, `perf_counter` should be used instead of `time`.
See https://stackoverflow.com/a/66037261
Also, to avoid one-time crashes, maybe we could run it a few times and compare the means (maybe use `timeit`?)